linker_script/sections/uninit_group.c++

```diff
@@ -52,7 +52,8 @@ UninitGroup::UninitGroup(const fdt &dtb, Memory logical_memory,
   heap.output_name = "heap";
 
   heap.add_command("PROVIDE( metal_segment_heap_target_start = . );");
-  heap.add_command(". = __heap_size;");
+  heap.add_command(". = DEFINED(__heap_max) ? LENGTH(ram) - ( . - ORIGIN(ram) "
```

Comment:
more reasonable to change ram to virtual_memory.name?

Reply:
Agreed, this is the right thing to do.

Reply:
what prompt to change to  virtual_memory and why this is better?

Reply:
`virtual_memory` is a parameter of the `UninitGroup`, and is an object of the `Memory` class, which means it knows its own `name`. When the linker script gets generated, if for some reason `virtual_memory` isn't named "ram", this script will fail.